utils/s2n_array.c

```diff
@@ -67,7 +59,7 @@ void *s2n_array_add(struct s2n_array *array)
         GUARD_PTR(s2n_array_enlarge(array, array->capacity * 2));
     }
 
-    void *element = (uint8_t *) array->elements + array->element_size * array->num_of_elements;
+    void *element = array->mem.data + array->element_size * array->num_of_elements;
```

Comment:
Not your fault but I'm surprised that we have implicit operator precedence here and not explicit parens.